h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -1108,23 +1115,37 @@ public TwoDimTable toTwoDimTable() {
    *  associated Chunks.
    *  @return Passed in Futures for flow-coding  */
   @Override public Futures remove_impl( Futures fs ) {
+    bulk_remove(new Key[]{_key}, nChunks());
+    return fs;
+  }
+
+  static void bulk_remove( final Key[] keys, final int ncs ) {
+    // Need to mark the Vec as mutating to make sure that no running computations of RollupStats will
+    // re-insert the rollups into DKV after they are deleted in bulk_remove(Key, int).
+    for (Key key : keys) setMutating(chunkKey(key, -2));
```

Comment:
For performance reasons, it would be better to call an asynchronous version taking Futures as an extra argument and blocking for all futures right after. This can be quite a few remote calls, better to do them in parallel.